packages/flame/benchmark/render_components_benchmark.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:async';
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:benchmark_harness/benchmark_harness.dart';
+import 'package:canvas_test/canvas_test.dart';
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+
+const _amountComponents = 10;
+const _amountTicks = 1000;
+
+class RenderComponentsBenchmark extends AsyncBenchmarkBase {
+  final Random random;
+
+  late final Canvas _canvas;
+  late final FlameGame _game;
+
+  RenderComponentsBenchmark(this.random) : super('Render Components Benchmark');
+
+  static Future<void> main() async {
+    final r = Random(69420);
+    await RenderComponentsBenchmark(r).report();
+  }
+
+  @override
+  Future<void> setup() async {
+    _canvas = MockCanvas();
+
+    _game = FlameGame();
+    _game.onGameResize(Vector2.all(100.0));
+
+    await _game.addAll(
+      List.generate(
```

Comment:
Later I guess we'll want a bit of deeper tree here

Reply:
this gets 3 to 5 levels according to my tests. we can easily control that by reducing the `2` factor

Reply:
I think it is good now: https://github.com/flame-engine/flame/pull/3438

Reply:
Nice!